### PR TITLE
kcov: 38 -> 41

### PR DESCRIPTION
**Pull request by tricktron**
###### Description of changes

Update kcov to v41.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by trofi** [comment]
Result of `nixpkgs-review pr 220464` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>cargo-kcov</li>
    <li>kcov</li>
  </ul>
</details>

**Review by trofi — APPROVED** [review]

File: pkgs/development/tools/analysis/kcov/default.nix [hunk 1]
```diff
@@ -10,34 +10,23 @@
...
         repo = "kcov";
         rev = "v${version}";
-        sha256 = "sha256-6LoIo2/yMUz8qIpwJVcA3qZjjF+8KEM1MyHuyHsQD38=";
+        sha256 = "sha256-Kit4Yn5Qeg3uAc6+RxwlVEhDKN6at+Uc7V38yhDPrAY=";
       };
 
-      patches = [
-        # Pull upstream patch for binutils-2/39 support:
-        #   https://github.com/SimonKagstrom/kcov/pull/383
-        (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/development/tools/analysis/kcov/default.nix:34]: Please remove the unused fetchpatch from inputs.

File: pkgs/development/tools/analysis/kcov/default.nix [hunk 2]
```diff
@@ -46,25 +35,25 @@ let
...
 
-        works-on-rust = runCommand "works-on-rust" {} ''
+        works-on-rust = runCommandCC "works-on-rust" { } ''
           set -ex
           cat - > a.rs <<EOF
           fn main() {}
           EOF
           # Put gcc in the path so that `cc` is found
-          PATH=${gcc}/bin:$PATH ${rustc}/bin/rustc a.rs -o a.out
+          ${rustc}/bin/rustc a.rs -o a.out
```
Comment (SuperSandro2000) [inline pkgs/development/tools/analysis/kcov/default.nix:56]: Please move to nativeBuildInputs

File: pkgs/development/tools/analysis/kcov/default.nix (hunk 1 above)
Comment (tricktron) [reply pkgs/development/tools/analysis/kcov/default.nix:34]: Done

File: pkgs/development/tools/analysis/kcov/default.nix (hunk 2 above)
Comment (tricktron) [reply pkgs/development/tools/analysis/kcov/default.nix]: Done

**Comment by tricktron** [comment]
@SuperSandro2000 Applied your review suggestions.

**Pull request merged**